app/controllers/orders_controller.rb

```diff
@@ -16,12 +16,16 @@ def create
       )
     )
     @order.save_card if current_user.credit_card.card_token.nil?
-    if @order.save_with_payment
-      @order.purchase_line_items
-      OrderMailer.receipt_email(@order.user).deliver_now
-      redirect_to order_path(@order), notice: "Your order has been completed"
+    if params[:preview]
+      render partial: "preview"
     else
-      redirect to cart_path(current_user), notice: "Your order could not be processed."
+      if @order.save_with_payment
+        @order.purchase_line_items
+        OrderMailer.receipt_email(@order.user).deliver_now
+        redirect_to order_path(@order), notice: "Your order has been completed"
+      else
+        redirect to cart_path(current_user), notice: "Your order could not be processed."
+      end
```

Comment:
So this method is starting to have a lot of responsibility and contains a lot of conditionals. It's definitely time for a refactor. If I may suggest:

```
class OrdersController < ApplicationController
class OrdersController < ApplicationController
  before_action :authenticate_user!
  before_action :set_order, only: [:show]

  def new
    @order = Order.new(user: current_user)
    @order.user.credit_card ||= @order.user.build_credit_card
  end

  def create
    @order = Order.new(
      order_params.merge(
        user: current_user,
        total: current_user.cart.total_cart_price,
        stripe_token: params[:stripeToken]
      )
    )
    prepare_order(@order)
  end

  private

  def set_order
    @order = Order.find(params[:id])
  end

  def prepare_order(order)
    save_card_to_order(order)
    render_order(order)
  end

  def save_card_to_order(order)
    order.save_card if current_user.credit_card.card_token.nil?
  end

  def render_order(order)
    if params[:preview]
      render partial: "preview"
    else
      save_order(order)
    end
  end

  def save_order(order)
    if order.save_with_payment
      handle_saved_order(order)
    else
      redirect to cart_path(current_user), notice: "Your order could not be processed."
    end
  end

  def handle_saved_order(order)
    order.purchase_line_items
    OrderMailer.receipt_email(order.user).deliver_now
    redirect_to order_path(order), notice: "Your order has been completed"
  end

  # strong params
  def order_params
    params.require(:order).permit(
      :user_id,
      :total,
      :stripe_token,
      :shipping_address1,
      :shipping_address2,
      :shipping_city,
      :shipping_state,
      :shipping_zip,
      credit_card_attributes: [
        :id,
        :billing_address1,
        :billing_address2,
        :billing_city,
        :billing_state,
        :billing_zip,
        :last_four_digits,
        :card_token
      ]
    )
  end
end
```

Everything still passes according to your Cucumber and RSpec tests. But I didn't do anything too special. 
1. I simply passed the instance variable as an argument so you don't have to keep typing `order` out as an instance variable.
2. Most importantly, I tried to decouple the responsibilities of your method in order to suit Sandi's :pray: Single Responsibility methodology.

This is definitely not the cleanest way to do this. You really should be handling this via services (don't worry about that for this project). I definitely believe this needs a refactor but don't feel like you have to take my suggested code.

If you have any questions about this, please ask!